packages/hoppscotch-cli/src/utils/checks.ts

```diff
@@ -0,0 +1,103 @@
+import fs from "fs/promises";
+import { CommanderError } from "commander";
+import { join, extname } from "path";
+import { pipe } from "fp-ts/function";
+import {
+  translateToNewRESTCollection,
+  isHoppRESTRequest,
+} from "@hoppscotch/data";
+import * as S from "fp-ts/string";
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import { error, HoppCLIError } from "../types";
+
+/**
+ * Typeguard to check valid Hoppscotch REST Collection
+ * @param param The object to be checked
+ * @returns Boolean value corresponding to the validity check
+ */
+export function isRESTCollection(param: any): E.Either<boolean, boolean> {
```

Comment:
`any` is a type which is really frowned upon.

Take the scenario where `null` or `undefined` or some other random value is passed as the `param` value., because of the any type, TypeScript will ignore type checking. Instead go for the `unknown` type. 

See: [https://mariusschulz.com/blog/the-unknown-type-in-typescript](https://mariusschulz.com/blog/the-unknown-type-in-typescript)